src/run.ts

```diff
@@ -29,6 +29,17 @@ export async function run() {
          required: true
       })
       const clusterName = core.getInput('cluster-name', {required: true})
+      const resourceTypeFleet = 'microsoft.containerservice/fleets'
+      const resourceTypeManagedCluster =
+         'microsoft.containerservice/managedclusters'
+      type ResourceType =
+         | typeof resourceTypeFleet
+         | typeof resourceTypeManagedCluster
+      const resourceInput = (
+         core.getInput('resource-type') ||
+         'Microsoft.ContainerService/managedClusters'
+      ).toLowerCase() as ResourceType
```

Comment:
we shouldn't need this type assertion if the narrowing happens before we assign to the ResourceType:

1. ResourceInput is read as a string
2.  We throw if it has values that are not either `typeof resourceTypeFleet` or `typeof resourceTypeCluster`
3.  Now the type is narrowed to fit the `type ResourceType` so we can directly assign without `as`